lib/Service/Html.php

```diff
@@ -129,11 +133,22 @@ public function sanitizeHtmlMailBody($mailBody, array $messageParameters, Closur
 
 		HTMLPurifier_URISchemeRegistry::instance()->register('cid', new CidURIScheme());
 
+
 		$purifier = new HTMLPurifier($config);
 
 		$result = $purifier->purify($mailBody);
 		// eat xml parse errors within HTMLPurifier
 		libxml_clear_errors();
+
+		// Add back the style tag
+		$styles = $purifier->context->get('StyleBlocks');
+		if ($styles) {
+			$result = implode("\n", array(
+				'<style type="text/css">',
```

Comment:
@LukasReschke anything scary about this? Is it safe? :)

Reply:
AFAIK it runs through CSSTidy, should be good :)

Reply:
There is a warning: http://htmlpurifier.org/live/configdoc/plain.html at "Filter.ExtractStyleBlocks". @LukasReschke imagecrash attack?

Reply:
according to @LukasReschke this change is okay as long as we're iframing the message.